velox/exec/tests/OrderByTest.cpp

```diff
@@ -516,3 +516,46 @@ TEST_F(OrderByTest, spillWithMemoryLimit) {
     OperatorTestBase::deleteTaskAndCheckSpillDirectory(task);
   }
 }
+
+TEST_F(OrderByTest, spillWithDecimal) {
+  constexpr int32_t kNumRows = 2000;
+  constexpr int64_t kMaxBytes = 1LL << 30; // 1GB
+  auto rowType =
+      ROW({"c0", "c1", "c2"},
+          {INTEGER(), SHORT_DECIMAL(10, 5), LONG_DECIMAL(30, 5)});
+  VectorFuzzer fuzzer({}, pool());
+  const int32_t numBatches = 5;
+  std::vector<RowVectorPtr> batches;
+  for (int32_t i = 0; i < numBatches; ++i) {
+    batches.push_back(fuzzer.fuzzRow(rowType));
+  }
+  auto tempDirectory = exec::test::TempDirectoryPath::create();
+  auto queryCtx = std::make_shared<core::QueryCtx>(executor_.get());
+  queryCtx->testingOverrideMemoryPool(
+      memory::defaultMemoryManager().addRootPool(
+          queryCtx->queryId(), kMaxBytes));
+  auto results =
+      AssertQueryBuilder(
+          PlanBuilder()
+              .values(batches)
+              .orderBy({fmt::format("{} ASC NULLS LAST", "c0")}, false)
+              .planNode())
+          .queryCtx(queryCtx)
+          .copyResults(pool_.get());
+  auto task =
+      AssertQueryBuilder(
+          PlanBuilder()
+              .values(batches)
+              .orderBy({fmt::format("{} ASC NULLS LAST", "c0")}, false)
+              .planNode())
+          .queryCtx(queryCtx)
+          .spillDirectory(tempDirectory->path)
+          .config(core::QueryConfig::kSpillEnabled, "true")
+          .config(core::QueryConfig::kOrderBySpillEnabled, "true")
+          .config(QueryConfig::kOrderBySpillMemoryThreshold, std::to_string(1))
+          .assertResults(results);
```

Comment:
We should create an `expectedResults` vector similar to how the input is created and verify the result.

Reply:
Didn't `.assertResults(results)` already do the result comparison?

Reply:
`results` are also computed above. A bug causing incorrect results will still satisfy this assert since the same incorrect results are being compared.
My suggestion is to manually create the expected results vector and use that to verify.

Reply:
> A bug causing incorrect results will still satisfy this assert since the same incorrect results are being compared.

That sounds like making the case not only test "spilling with decimal type" but also "querying with decimal type". So I guess we can make a sperate case for the latter one?

Reply:
Adding some backgrounds:

The error was only thrown when spilling is engaged. OrderBy was working well with decimal type without spilling.

Reply:
@zhztheplayer Thanks for the clarification. This test works then.